app/components/retriable_fast_image/index.tsx

```diff
@@ -30,12 +43,24 @@ export default class RetriableFastImage extends PureComponent<RetriableFastImage
     }
 
     render() {
-        return (
-            <FastImage
-                {...this.props}
-                key={`${this.props.id}-${this.state.retry}`}
-                onError={this.onError}
-            />
-        );
+        const valid = this.props.source && typeof this.props.source != 'number' && this.props.source.uri ? validator.isURL(this.props.source.uri, this.config) || validator.isDataURI(this.props.source.uri) : false;
+        if (valid) {
+            return (
+                <FastImage
+                    key={`${this.props.id}-${this.state.retry}`}
+                    onError={this.onError}
+                    {...this.props}
+                />
+            );
+        }
+
+        if (this.props.renderOnError) {
+            return (
+                <CompassIcon
+                    name='jumbo-attachment-image-broken'
```

Comment:
Don't you need to set a size?
cc: @matthewbirtch

Reply:
> Don't you need to set a size?
> cc: @matthewbirtch

I'm afraid I don't know what size to propose here since I don't know the context of where this is appearing. Could you help with the context?

Reply:
@enahum I've updated all the components to show the renderOnError.

@matthewbirtch - anywhere we call `<FastImage />` to display an image in the app, if it fails it's going to show the `jumbo-attachment-image-broken` image, and I haven't given it a size; I thought I'd leave it undefined/unstyled as it will default to it's icon size, or shrink down for smaller image's (e.g. icons, emoji's etc) to fit in the styling defined in the `<FastImage />`

Reply:
@shazm could you provide some screenshot examples? In the meantime I'll get a build going and see if I can see it for myself.